beprof/profile.py

```diff
@@ -144,11 +146,14 @@ def normalize(self, dt):
         except RuntimeWarning as e:
             logger.error('in normalize(). self class is {0}, dt={1}'.format(self.__class__, dt))
             raise Exception("Scaling factor error:\n" + str(e))
-        self.y /= ave
+        self.y = self.y / ave
```

Comment:
You can keep this change, but add info in docstring what happens if you have array of integers, explaining what if you:
 - normalize with integer factor
 - normalize with floating-point factor

But before proceeding further: consider these two snippets: `self.y /= ave` and `self.y = self.y / ave` - which one requires additional memory allocation ?

What if `self.y` is large array or floats (say 1e6) and we normalize by factor 0.5. Which one of abovementioned snippets will run faster ?

Reading https://docs.scipy.org/doc/numpy/reference/internals.html might help.